libcalico-go/lib/set/diff.go

```diff
@@ -16,16 +16,30 @@ package set
 
 // IterDifferences iterates through the set of items that are in A but not in B, and the set that are in B but not in A.
 func IterDifferences[T comparable](a, b Set[T], aNotB, bNotA func(item T) error) {
-	a.Iter(func(item T) error {
+	for item := range a.All() {
```

Comment:
Leave these ones as they were?  The code is definitely more complicated now, and IIUC it is still valid to use the `.Iter()` form.